lib/icingadb/redisconnection.cpp

```diff
@@ -412,13 +412,19 @@ void RedisConnection::ReadLoop(asio::yield_context& yc)
 							} catch (const boost::coroutines::detail::forced_unwind&) {
 								throw;
 							} catch (...) {
-								promise.set_exception(std::current_exception());
+								try {
+									promise.set_exception(std::current_exception());
+								} catch (...) {
+								}
 
 								continue;
 							}
 						}
 
-						promise.set_value(std::move(replies));
+						try {
+							promise.set_value(std::move(replies));
+						} catch (...) {
+						}
```

Comment:
Tell me which one and I'll change this in an instant.

Reply:
> But I would actually just add a bool here that states if the promise is already failed and only set the promise depending on that bool.

Well this would be my preferred change.

Reply:
https://stackoverflow.com/questions/12265451/ask-forgiveness-not-permission-explain

Reply:
Well, that's one of these Python idioms where I don't understand why this is supposed to be a good idea.

So yeah, sure, there are cases where you want to catch as much as you can. But I wouldn't do this in places where the exception is then silently ignored as this has the risk to hide bugs and make debugging harder. I mean image someone introduces a bug where [a `future_error` with code `no_state`](https://en.cppreference.com/w/cpp/thread/promise/set_value#Exceptions) is thrown, I guess they'd have lots of fun debugging when this exception is silently swallowed here.

Reply:
> Well, that's one of these Python idioms where I don't understand why this is supposed to be a good idea.

Think of file exists ok, then open errors vs. just open.

Don’t misunderstand me: I actually prefer to catch only x **if** only x is on my way **and** I can define x. Oh, I've overseen the Exceptions section there. 🐘😅